### PR TITLE
store/tikv: implement a `ttlManager` to update the TTL of a transaction (#12177)

### PR DESCRIPTION
**Pull request by tiancaiamao**
Cherry-pick https://github.com/pingcap/tidb/pull/12177 and https://github.com/pingcap/tidb/pull/12398

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @lysu @MyonKeminta

**Comment by tiancaiamao** [comment]
PTAL @coocood @lysu @MyonKeminta

**Comment by coocood** [comment]
LGTM

File: metrics/metrics.go [hunk 1]
```diff
@@ -151,5 +151,7 @@ func RegisterMetrics() {
 	prometheus.MustRegister(TiKVRangeTaskStats)
 	prometheus.MustRegister(TiKVRangeTaskPushDuration)
 	prometheus.MustRegister(HandleSchemaValidate)
+	prometheus.MustRegister(TiKVTokenWaitDuration)
```
Comment (MyonKeminta) [inline metrics/metrics.go]: Is this metric used? Why is it picked into this PR?

Comment (tiancaiamao) [reply metrics/metrics.go]: Done @MyonKeminta

**Pull request merged**